# This file is part of HiGFXback

# requires
REQUIRES="libevdev-build libpng-build meson-build tiff-build"

PKG_CONFIG_PATH=/drm/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

if PKG_CONFIG_PATH=/drm/share/pkgconfig pkg-config --exists egl-opengl-stubs-build libdrm-build libgbm-stub-build; then
  GL=1
  EGL_OPENGL="-Degl-fbdev=false -Dgl=false -Dglesv1_cm=false"
  REQUIRES="$REQUIRES egl-opengl-stubs-build libdrm-build libgbm-stub-build"
fi

if PKG_CONFIG_PATH=/drm/share/pkgconfig pkg-config --exists glslang-build vulkan-loader-build; then
  VK=1
  REQUIRES="$REQUIRES glslang-build vulkan-loader-build"
fi

test $GL || test $VK || { echo OpenGL or Vulkan packages required; exit 1; }

if PKG_CONFIG_PATH=/drm/share/pkgconfig pkg-config --exists sdl2-build; then
  test $VK && PKG_CONFIG_PATH=/drm/share/pkgconfig pkg-config --print-requires sdl2-build | grep -q vulkan-loader-build || { echo SDL2 with Vulkan support required; exit 1; }
  SDL2=1
  REQUIRES="$REQUIRES sdl2-build"
fi

if PKG_CONFIG_PATH=/drm/share/pkgconfig pkg-config --exists sfml-build; then
  SFML=1
  REQUIRES="$REQUIRES sfml-build"
fi

# configure
PKG_CONFIG_PATH=/drm/lib/pkgconfig meson setup $EGL_OPENGL --prefix=/drm build

# build
meson compile -C build

# install
install -d $DESTDIR/drm/bin
test $GL && install build/yagears2 $DESTDIR/drm/bin
test $GL && test $SDL2 || test $SFML && install build/yagears2-gui $DESTDIR/drm/bin
test $VK && install build/yagears2-vk $DESTDIR/drm/bin
test $VK && test $SDL2 && install build/yagears2-vk-gui $DESTDIR/drm/bin

# build.pc
install -d $DESTDIR/drm/share/pkgconfig
cat > $DESTDIR/drm/share/pkgconfig/yagears2-build.pc << EOF
Name: yagears
Version: 2.8
Description: Yet Another Gears OpenGL / Vulkan demo
Requires: $REQUIRES

exec=\\
EOF
test $GL && echo /drm/bin/yagears2 \\ >> $DESTDIR/drm/share/pkgconfig/yagears2-build.pc
test $GL && test $SDL2 || test $SFML && echo /drm/bin/yagears2-gui \\ >> $DESTDIR/drm/share/pkgconfig/yagears2-build.pc
test $VK && echo /drm/bin/yagears2-vk \\ >> $DESTDIR/drm/share/pkgconfig/yagears2-build.pc
test $VK && test $SDL2 && echo /drm/bin/yagears2-vk-gui \\ >> $DESTDIR/drm/share/pkgconfig/yagears2-build.pc
sed -i '$ s/ \\//' $DESTDIR/drm/share/pkgconfig/yagears2-build.pc

exit
--- yagears-2.8.orig/CMakeLists.txt
+++ yagears-2.8/CMakeLists.txt
@@ -710,7 +710,7 @@
   set(GUI OFF)
 endif()
 
-if((ENABLE_VK_X11 OR ENABLE_VK_DIRECTFB OR ENABLE_VK_FBDEV OR ENABLE_VK_WAYLAND OR ENABLE_VK_XCB OR ENABLE_VK_D2D) AND (ENABLE_GLFW OR (ENABLE_SDL AND WITH_SDL STREQUAL 2)))
+if((ENABLE_VK_X11 OR ENABLE_VK_DIRECTFB OR ENABLE_VK_FBDEV OR ENABLE_VK_WAYLAND OR ENABLE_VK_XCB OR ENABLE_VK_D2D) AND (ENABLE_GLFW OR ENABLE_SDL AND WITH_SDL STREQUAL 2))
   set(VK_GUI ON)
 else()
   set(VK_GUI OFF)
--- yagears-2.8.orig/meson.build
+++ yagears-2.8/meson.build
@@ -701,7 +701,7 @@
   GUI = false
 endif
 
-if (enable_vk_x11 or enable_vk_directfb or enable_vk_fbdev or enable_vk_wayland or enable_vk_xcb or enable_vk_d2d) and (enable_glfw or (enable_sdl and with_sdl == '2'))
+if (enable_vk_x11 or enable_vk_directfb or enable_vk_fbdev or enable_vk_wayland or enable_vk_xcb or enable_vk_d2d) and (enable_glfw or enable_sdl and with_sdl == '2')
   VK_GUI = true
 else
   VK_GUI = false
